internal/tiltfile/live_update.go

```diff
@@ -181,66 +221,47 @@ func (s *tiltfileState) liveUpdateStepToModel(l liveUpdateStep) (model.LiveUpdat
-}
-
-func (s *tiltfileState) consumeLiveUpdateStep(stepToConsume liveUpdateStep) {
-	for i, step := range s.unconsumedLiveUpdateSteps {
-		if step.declarationPosition() == stepToConsume.declarationPosition() {
-			copy(s.unconsumedLiveUpdateSteps[i:], s.unconsumedLiveUpdateSteps[i+1:])
-			s.unconsumedLiveUpdateSteps = s.unconsumedLiveUpdateSteps[:len(s.unconsumedLiveUpdateSteps)-1]
-			break
-		}
-	}
-}
-
-func (s *tiltfileState) liveUpdate(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	var dockerRef string
-	var steps, fullRebuildTriggers starlark.Value
-	if err := starlark.UnpackArgs(fn.Name(), args, kwargs,
-		"image", &dockerRef,
-		"steps", &steps,
-		"full_rebuild_triggers?", &fullRebuildTriggers,
-	); err != nil {
-		return nil, err
-	}
-
+func (s *tiltfileState) liveUpdateFromSteps(maybeSteps starlark.Value) (model.LiveUpdate, error) {
 	var modelSteps []model.LiveUpdateStep
-	stepSlice := starlarkValueOrSequenceToSlice(steps)
+	stepSlice := starlarkValueOrSequenceToSlice(maybeSteps)
+	var fallBackOn []string
+
 	for _, v := range stepSlice {
 		step, ok := v.(liveUpdateStep)
 		if !ok {
-			return starlark.None, fmt.Errorf("'steps' must be a list of live update steps - got value '%v' of type '%s'", v.String(), v.Type())
+			return model.LiveUpdate{}, fmt.Errorf("'steps' must be a list of live update steps - got value '%v' of type '%s'", v.String(), v.Type())
 		}
 
 		ms, err := s.liveUpdateStepToModel(step)
 		if err != nil {
-			return starlark.None, err
+			return model.LiveUpdate{}, err
 		}
 		s.consumeLiveUpdateStep(step)
-		modelSteps = append(modelSteps, ms)
-	}
 
-	frtSlice := starlarkValueOrSequenceToSlice(fullRebuildTriggers)
-	var frtStrings []string
-	for _, v := range frtSlice {
-		str, ok := v.(starlark.String)
-		if !ok {
-			return starlark.None, fmt.Errorf("'full_rebuild_triggers' must only contain strings - got value '%v' of type '%s'", v.String(), v.Type())
+		// HACK(maia): temporary hack for backwards compatibility/making this a smaller PR--
+		// use the existing LiveUpdate constructor. (Soon, we'll treat fall_back_on as
+		// just another step INTERNALLY too (instead of as a non-step property of a LiveUpdate).
+		if fallBackStep, ok := ms.(model.LiveUpdateFallBackOnStep); ok {
+			fallBackOn = append(fallBackOn, fallBackStep.Files...)
+		} else {
+			modelSteps = append(modelSteps, ms)
 		}
-		frtStrings = append(frtStrings, string(str))
 	}
 
-	s.liveUpdates[dockerRef] = &liveUpdateDef{
-		steps:               modelSteps,
-		fullRebuildTriggers: frtStrings,
-	}
+	return model.NewLiveUpdate(modelSteps, model.NewPathSet(fallBackOn, s.absWorkingDir()))
+}
 
-	return starlark.None, nil
+func (s *tiltfileState) consumeLiveUpdateStep(stepToConsume liveUpdateStep) {
+	for i, step := range s.unconsumedLiveUpdateSteps {
+		if step.declarationPosition() == stepToConsume.declarationPosition() {
```

Comment:
fwiw, i'm not at all convinced that this equality check will behave like you expect. it has unexported pointer fields. and the struct might change a lot in the future

it might be more robust to create your own struct with the equality semantics you want, and use it as a map key to do inserts/deletes